internal/vm/importers/glob.go

```diff
@@ -0,0 +1,289 @@
+package importers
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-jsonnet"
+	"github.com/pkg/errors"
+)
+
+const (
+	globParamDirs           = "dirs"
+	globParamVerb           = "verb"
+	globParamStripExtension = "strip-extension"
+)
+
+var supportedParams = map[string]bool{
+	globParamDirs:           true,
+	globParamVerb:           true,
+	globParamStripExtension: true,
+}
+
+var supportedParamsDisplay string
+
+func init() {
+	var ret []string
+	for k := range supportedParams {
+		ret = append(ret, k)
+	}
+	sort.Strings(ret)
+	supportedParamsDisplay = strings.Join(ret, ", ")
+}
+
+// globImportParams are additional parameters for the glob import expressed as URI query params.
+type globImportParams struct {
+	verb           string // one of 'import' or 'importstr' - the verb to use to import files found
+	dirs           int    // number of directories levels to retain in the returned object (default: all)
+	stripExtension bool   // whether extensions should be stripped from the base name of the file in the map key
+}
+
+// keyFor returns the key to be used for the supplied file based on whether directory levels should be limited
+// and/ or extensions stripped.
+func (p globImportParams) keyFor(file string) string {
+	// short-circuit simple case
+	if p.dirs < 0 && !p.stripExtension {
```

Comment:
Nit: This if seems like a bug magnet to me. This is just a special case, which is handled by the rest of the code. If another option is added it will be very easy to forget about this one. The only advantage I see is avoiding additional computation (which is trivial compared to the IO involved with importing anything or even parsing the output that is produced here).